providers/implementations/signature/rsa_sig.c

```diff
@@ -1541,3 +1931,272 @@ const OSSL_DISPATCH ossl_rsa_signature_functions[] = {
+                                                                        \
+    const OSSL_DISPATCH ossl_rsa_##md##_signature_functions[] = {       \
+        { OSSL_FUNC_SIGNATURE_NEWCTX, (void (*)(void))rsa_newctx },     \
+        { OSSL_FUNC_SIGNATURE_SIGN_INIT,                                \
+          (void (*)(void))rsa_##md##_sign_init },                       \
+        { OSSL_FUNC_SIGNATURE_SIGN, (void (*)(void))rsa_sign },         \
+        { OSSL_FUNC_SIGNATURE_SIGN_MESSAGE_INIT,                        \
+          (void (*)(void))rsa_##md##_sign_message_init },               \
+        { OSSL_FUNC_SIGNATURE_SIGN_MESSAGE_UPDATE,                      \
+          (void (*)(void))rsa_signverify_message_update },              \
+        { OSSL_FUNC_SIGNATURE_SIGN_MESSAGE_FINAL,                       \
+          (void (*)(void))rsa_sign_message_final },                     \
+        { OSSL_FUNC_SIGNATURE_VERIFY_INIT,                              \
+          (void (*)(void))rsa_##md##_verify_init },                     \
+        { OSSL_FUNC_SIGNATURE_VERIFY,                                   \
+          (void (*)(void))rsa_verify },                                 \
+        { OSSL_FUNC_SIGNATURE_VERIFY_MESSAGE_INIT,                      \
+          (void (*)(void))rsa_##md##_verify_message_init },             \
+        { OSSL_FUNC_SIGNATURE_VERIFY_MESSAGE_UPDATE,                    \
+          (void (*)(void))rsa_signverify_message_update },              \
+        { OSSL_FUNC_SIGNATURE_VERIFY_MESSAGE_FINAL,                     \
+          (void (*)(void))rsa_verify_message_final },                   \
+        { OSSL_FUNC_SIGNATURE_VERIFY_RECOVER_INIT,                      \
+          (void (*)(void))rsa_##md##_verify_recover_init },             \
+        { OSSL_FUNC_SIGNATURE_VERIFY_RECOVER,                           \
+          (void (*)(void))rsa_verify_recover },                         \
+        { OSSL_FUNC_SIGNATURE_FREECTX, (void (*)(void))rsa_freectx },   \
+        { OSSL_FUNC_SIGNATURE_DUPCTX, (void (*)(void))rsa_dupctx },     \
+        { OSSL_FUNC_SIGNATURE_QUERY_KEY_TYPES,                          \
+          (void (*)(void))rsa_sigalg_query_key_types },                 \
+        { OSSL_FUNC_SIGNATURE_GET_CTX_PARAMS,                           \
+          (void (*)(void))rsa_get_ctx_params },                         \
+        { OSSL_FUNC_SIGNATURE_GETTABLE_CTX_PARAMS,                      \
+          (void (*)(void))rsa_gettable_ctx_params },                    \
+        { OSSL_FUNC_SIGNATURE_SET_CTX_PARAMS,                           \
+          (void (*)(void))rsa_sigalg_set_ctx_params },                  \
+        { OSSL_FUNC_SIGNATURE_SETTABLE_CTX_PARAMS,                      \
+          (void (*)(void))rsa_sigalg_settable_ctx_params },             \
+        OSSL_DISPATCH_END                                               \
+    }
+
+#if 0                            /* Linking problems */
```

Comment:
These algorithms which depend on legacy hashes should be implemented in the legacy provider if anywhere.

Reply:
Yeah I agree...  but there are some "interesting" side effects with that.

Reply:
I've done what @slontis suggested, just removed them.